### DISCUSSION TITLE
French morphologizer mislabeling future, conditional, imperative

### DISCUSSION THREAD

**Discussion by KennethBaclawskiJrDialpad** [explosion/spaCy#13717] [Language Support]
Hello,	

I'm using spaCy to model French conversations, and I see that the morphologizer is not performing as well as I'd expect for unambiguous irrealis forms (specifically future tense, conditional mood, imperatives). I understand the underlying reason is probably that these forms aren't frequent in the training data, but are there any potential updates or recommended workarounds?

For example:
* "Remplacez le verbe par un nom.": The imperative "Remplacez" is misrecognized as a noun (using `POS/TAG`). This is similar to #8147, but broader in that "Remplacez" is unambiguously a verb.
* "Vous remplacerez le verbe par un nom.": The future tense "remplacerez" is misrecognized as an imperative (`MORPH` contains `Mood=Imp`, `Tense=Pres`), even though it is unambiguously the future.
* "Vous remplaceriez le verbe par un nom.": The conditional mood "remplaceriez" is misrecognized as the future tense (`MORPH` contains `Mood=Ind`, `Tense=Fut`), even though it is unambiguously the conditional.

I'm working with a set of ~160 common French verbs and tested their whole paradigms in this way. 98% of infinitives are recognized correctly, but only 13% of second person plural imperatives (34% even had incorrect POS like in the example above), 37% of future tense, and 7% of conditional mood forms. Sure enough, I see that these three categories are uncommon in the UD French Sequoia data.

### How to reproduce the behaviour

```
import spacy
nlp = spacy.load("fr_core_news_sm")
doc = nlp("Remplacez le verbe par un nom.")

for token in doc:
    print(token.text, token.pos_, token.morph)
```
### Info about spaCy

spaCy version: 3.8.2
Python version: 3.11.9
Pipelines: fr_core_news_sm (3.8.0)

**Comment by thjbdvlt** [comment]
You could try the pipeline i've trained, which is available here: https://github.com/thjbdvlt/solipCysme

I've made it because I have the same issue you. It's trained mostly on novels (19e-21e) and texts with a lot of interactions, personnal pronouns and differents moods. The morphologizer uses [HunSpell](http://hunspell.github.io/) output (`po:` and `is:`) as features (along with `PREFIX` and `NORM`), so it should have good results even on texts very different that the ones my model has been trained on.

It has some limits. First, there is no `Gender` morphological feature, so it would have terrible scores on (e. g.) UD datasets. It's not absolutely accurate on `NOUN`/`ADJ` for sentences like *les jeunes bourgeois* (where both words could be `ADJ`/`NOUN`). And some other stuff (like `ADJ`/Passive Past Participe; or "que" as `PRON` and "que" as `SCONJ`).

Of course I could provide some information if you need it. Let me know!
